zowe/zowe-cli
# feat(cfg): Schema management using `ProfileInfo`

**What It Does**

This PR implements:
- **Schema management** APIs in Imperative's `ProfileInfo` class, primarily to allow schema contributions from extenders and plug-ins of Zowe client applications. Also provides functionality for the following use cases:
  - _Retrieve the schema for a specific profile type_
  - _Rebuild the entire schema using all available profile types, or rebuild the schema with types sourced from a specific application_
  - _Retrieve the list of sources for a specific profile type_
  - _Add a profile type to a layer in the configuration file (with defaults, if any exist)_
- The concept of **schema versioning**, enabling the following:
  - _Replace a schema on-disk with a newer version, if provided_
  - _Recognize conflicts between schemas based on their versions_
  - _Warn users when a schema has been upgraded to a major version_
  - _Warn users if an extension/plug-in tries to register with an older major schema version than the one on-disk_

**How to Test**

Create a test plug-in or extension that uses `ProfileInfo` and try one (or more) of the newly-implemented APIs:
- `addProfileToConfig`
- `addProfileTypeToSchema`
- `getProfileTypes`
- `buildSchema`
- `getSchemaForType`

Alternatively, build and run tests, then do black-box testing to determine whether default behavior has remained the same. Testers should not notice any obvious differences when using the CLI build in this PR.

**Review Checklist**
I certify that I have:
